python/ray/experimental/sgd/tf/examples/cifar_tf_example.py

```diff
@@ -182,46 +185,117 @@ def data_augmentation_creator(config):
     test_size = 10000
     batch_size = args.batch_size
 
-    num_train_steps = 10 if args.smoke_test else data_size // batch_size
-    num_eval_steps = 10 if args.smoke_test else test_size // batch_size
+    num_train_steps = (10 if args.smoke_test else data_size // batch_size)
+    num_eval_steps = (10 if args.smoke_test else test_size // batch_size)
+
+    def init_hook():
+        os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+        os.environ["TF_CPP_MIN_VLOG_LEVEL"] = "3"
+
+        import tensorflow as tf
+
+        gpus = tf.config.experimental.list_physical_devices("GPU")
+        if len(gpus) == 0:
+            if args.use_gpu:
+                raise "No GPUs found!"
+
+        import logging
+        tf.get_logger().setLevel(logging.ERROR)
+
+        if not args.use_gpu:
+            tf.keras.backend.set_image_data_format("channels_last")
+        else:
+            tf.keras.backend.set_image_data_format("channels_first")
+
+    init_hook()
+    (x_train, y_train), (x_test, y_test) = fetch_keras_data()
+
+    config = {
+        "batch_size": batch_size,
+        "fit_config": {
+            "steps_per_epoch": num_train_steps,
+        },
+        "evaluate_config": {
+            "steps": num_eval_steps,
+        },
+        "input_shape": x_train.shape[1:]
+    }
 
     trainer = TFTrainer(
         model_creator=create_model,
         data_creator=(data_augmentation_creator
                       if args.augment_data else data_creator),
+        init_hook=init_hook,
         num_replicas=args.num_replicas,
         use_gpu=args.use_gpu,
         verbose=True,
-        config={
-            "batch_size": batch_size,
-            "fit_config": {
-                "steps_per_epoch": num_train_steps,
-            },
-            "evaluate_config": {
-                "steps": num_eval_steps,
-            }
-        })
-
-    training_start = time.time()
-    for i in range(3):
-        # Trains num epochs
-        train_stats1 = trainer.train()
-        train_stats1.update(trainer.validate())
-        print("iter {}:".format(i), train_stats1)
-
-    dt = (time.time() - training_start) / 3
-    print(f"Training on workers takes: {dt:.3f} seconds/epoch")
-
-    model = trainer.get_model()
-    trainer.shutdown()
-    dataset, test_dataset = data_augmentation_creator(
-        dict(batch_size=batch_size))
-
-    training_start = time.time()
-    model.fit(dataset, steps_per_epoch=num_train_steps, epochs=1)
-    dt = (time.time() - training_start)
-    print(f"Training on workers takes: {dt:.3f} seconds/epoch")
-
-    scores = model.evaluate(test_dataset, steps=num_eval_steps)
-    print("Test loss:", scores[0])
-    print("Test accuracy:", scores[1])
+        config=config)
+
+    if args.time_only:
```

Comment:
Should this be a separate script? 

The cifar tf example should be a dead simple example that works out of the box (without bells and whistles).